lightkurve/targetpixelfile.py

```diff
@@ -1126,12 +1126,24 @@ def _open_image(img, extension):
 
             if idx == 0:  # Get default keyword values from the first image
                 factory.keywords = hdu.header
+
+
+            # Get positional shift of the image compared to the reference WCS
+            wcs_current = WCS(hdu.header)
+            column_current, row_current = wcs_current.wcs_world2pix(
+                          np.asarray([[position.ra.deg], [position.dec.deg]]).T,0)[0]
+            column_ref, row_ref = wcs_ref.wcs_world2pix(
+                          np.asarray([[position.ra.deg], [position.dec.deg]]).T,0)[0]
+
+            hdu.header['POS_CORR1'] = column_current - column_ref
+            hdu.header['POS_CORR2'] = row_current - row_ref
+
             if position is None:
                 cutout = hdu
             else:
                 cutout = Cutout2D(hdu.data, position, wcs=WCS(hdu.header),
                                   size=size, mode='partial')
-            factory.add_cadence(frameno=idx, flux=cutout.data, header=hdu.header)
+            factory.add_cadence(frameno=idx, wcs=wcs_ref, flux=cutout.data, header=hdu.header)
```

Comment:
I don't understand why the `add_cadence` method has a `wcs` argument.  All it seems to be used for inside `add_cadence` is this snippet:
```Python
        if wcs is None:
            self.pos_corr1[frameno], self.pos_corr2[frameno] = None, None
```

I suggest we clean this up and remove the `wcs` argument from the `add_cadence` method.  Inside that method, we can just do e.g. this:
```Python
        if 'POS_CORR1' in header:
            self.pos_corr1[frameno] = header['POS_CORR1']
        else:
            self.pos_corr1[frameno] = None
```

Reply:
@barentsen yes I was struggling over this same thing myself- why is the wcs fed in the places where it is, and why doesn't it go through to the TPF in the end? Maybe when you are free we can take a look together when you get back...